### PR TITLE
feat(helm): allow usage of additional env vars / configmap / secrets

### PR DESCRIPTION
**Pull request by MickaelCa**
This PR makes it possible to load external environmenent variables.
On my cluster, I'm using [Postgres Operator](https://access.crunchydata.com/documentation/postgres-operator/latest/) by CrunchyData, I need to be able to load the generated database secrets directly rather than copying it in the values.

### CODE REVIEW

File: dev/helm/values.yaml [hunk 1]
```diff
@@ -98,6 +98,22 @@ tolerations: []
...
+#      value: "sample value"
+#    - name: SAMPLE_ENV_FROM_SECRET
+#      valueFrom:
+#        secretKeyRef:
+#          name: 'secret-name'
+#          key: 'secret-key'
+  secrets: []
+#    - secret-name
+#    - another-secret-name
+  configmaps: []
```
Comment (bdsoha) [inline dev/helm/values.yaml:113]: IMO, this should be camelCase, as it is found in the k8s API *(`configMapRef`)*.

```suggestion
  configMaps: []
```

File: dev/helm/templates/deployment.yaml [hunk 2]
```diff
@@ -71,6 +75,19 @@ spec:
             {{- end }}
             - name: HA_ACTIVE
               value: {{ .Values.replicaCount | int | le 2 | quote }}
+          envFrom:
+            {{- with .Values.extraEnv.configmaps }}
```
Comment (bdsoha) [inline dev/helm/templates/deployment.yaml:79]:
```suggestion
            {{- with .Values.extraEnv.configMaps }}
```

**Review by bdsoha — CHANGES_REQUESTED** [review]
IMO, this should be camelCase, as it is found in the k8s API *(`configMapRef`)*.

File: dev/helm/templates/deployment.yaml [hunk 3]
```diff
@@ -39,14 +39,18 @@ spec:
...
             - name: DB_TYPE
               value: postgres
             {{- if (.Values.externalPostgresql).databaseURL }}
             - name: DATABASE_URL
               value: {{ .Values.externalPostgresql.databaseURL }}
             - name: NODE_TLS_REJECT_UNAUTHORIZED
               value: {{ default "1" .Values.externalPostgresql.NODE_TLS_REJECT_UNAUTHORIZED | quote }}
-            {{- else }}
+            {{- end }}
+            {{- if .Values.postgresql.enabled}}
```
Comment (stavros-k) [inline dev/helm/templates/deployment.yaml:53]:
```suggestion
            {{- if .Values.postgresql.enabled }}
```

File: dev/helm/templates/deployment.yaml [hunk 4]
```diff
@@ -39,14 +39,18 @@ spec:
           image: "{{ .Values.image.repository }}:{{ default "latest" .Values.image.tag }}"
           imagePullPolicy: {{ default "IfNotPresent" .Values.image.imagePullPolicy }}
           env:
+            {{- with .Values.extraEnv.env }}
```
Comment (stavros-k) [inline dev/helm/templates/deployment.yaml:42]: I'd suggest checking if `extraEnv` is not null first.
On upgrade, it won't exist in values.yaml and will most likely error with `{}.env` `nil` pointer.

**Pull request closed without merging**